plugins/worker/lib/BlockedBy.js

```diff
@@ -154,176 +165,199 @@ async function checkMultipleBuildsInSameEvent(buildConfig, blockingBuildIds, bui
-                    buildId,
-                    status: 'COLLAPSED',
-                    statusMessage: `Collapsed to build: ${lastRunningBuildId}`
-                })
-                .catch(err => {
-                    logger.error(`Failed to update build status to COLLAPSED for build:${buildId}:${err}`);
-                });
-            await this.queueObject.connection.redis.hdel(`${queuePrefix}buildConfigs`, buildId);
+        logger.info('%s | %s | Delete runningKey and waitingKey', buildId, jobId);
 
-            logger.info('%s | %s | Remove waiting key and collapse build', buildId, jobId);
+        // Should not proceed since this build was previously aborted
+        return false;
+    }
 
-            return false;
-        }
+    let blockedBy = this.args[0].blockedBy.split(',').map(jid => `${runningJobsPrefix}${jid}`);
 
-        // If this build is in the delete list (it was aborted)
-        if (shouldDelete !== null) {
-            await this.queueObject.connection.redis.del(deleteKey);
+    if (!enforceBlockedBySelf) {
+        blockedBy = blockedBy.filter(key => key !== `${runningJobsPrefix}${jobId}`); // remove itself from blocking list
+    }
 
-            //  Clean up to prevent race condition: stop and beforePerform happen at the same time
-            //  stop deletes key runningKey and waitingKey
-            //  beforePerform either proceeds or reEnqueue (which adds the key back)
-            await this.queueObject.connection.redis.lrem(waitingKey, 0, buildId);
+    if (blockedBy.length > 0) {
+        logger.info('%s | %s | BlockedBy list:%s', buildId, jobId, blockedBy);
 
-            if (parseInt(runningBuildId, 10) === buildId) {
-                await this.queueObject.connection.redis.del(runningKey);
-            }
+        const blockingBuildIds = [];
 
-            logger.info('%s | %s | Delete runningKey and waitingKey', buildId, jobId);
+        // Get the blocking job
+        await Promise.all(
+            blockedBy.map(async key => {
+                const val = await this.queueObject.connection.redis.get(key);
 
-            // Should not proceed since this build was previously aborted
-            return false;
-        }
+                if (val !== null) {
+                    blockingBuildIds.push(val);
+                }
+            })
+        );
 
-        let blockedBy = this.args[0].blockedBy.split(',').map(jid => `${runningJobsPrefix}${jid}`);
+        logger.info('%s | %s | blockingBuildIds:%s', buildId, jobId, blockingBuildIds);
 
-        if (!enforceBlockedBySelf) {
-            blockedBy = blockedBy.filter(key => key !== `${runningJobsPrefix}${jobId}`); // remove itself from blocking list
-        }
+        // If any blocking job is running, then re-enqueue
+        if (blockingBuildIds.length > 0) {
+            // if build is from same event then don't re-enqueue
+            const isSameBuild = await checkMultipleBuildsInSameEvent(
```

Comment:
This is probably no longer required, we can get rid of this? @jithine  Thoughts?

Reply:
agree, this should not be needed anymore.